src/Corax/Pipeline/IFilter.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Corax.Pipeline
+{
+    public interface IFilter : IDisposable
+    {
+        int Filter(ReadOnlySpan<byte> source, ref Span<Token> tokens);
```

Comment:
What is the return value here?

Reply:
The amount of source characters processed. I have been wary of removing it altogether, because this is essentially required if we are going to be doing multistep transformations. Will become `void` as soon as we know for sure we would not need it.

Reply:
What do you mean, multi step?

Reply:
If the input is bigger than the actual buffer. We are no longer having to deal with that as we know the max size if `1024` or `MaxTermsLength` now.